src/pybind/mgr/cephadm/tests/test_cephadm.py

```diff
@@ -2139,3 +2139,41 @@ def test_tuned_profiles_validation(self, cephadm_module):
                 cephadm_module.inventory.all_specs = mock.Mock(
                     return_value=[mock.Mock().hostname, mock.Mock().hostname])
                 cephadm_module._validate_tuned_profile_spec(spec)
+
+    def test_inventory_known_hostnames(self, cephadm_module):
+        cephadm_module.inventory.add_host(HostSpec('host1', '1.2.3.1'))
+        cephadm_module.inventory.add_host(HostSpec('host2', '1.2.3.2'))
+        cephadm_module.inventory.add_host(HostSpec('host3.domain', '1.2.3.3'))
+        cephadm_module.inventory.add_host(HostSpec('host4.domain', '1.2.3.4'))
+
+        # update_known_hostname expects args to be <hostname, shortname, fqdn>
+        # as are gathered from cephadm gather-facts. Although, passing the
+        # names in the wrong order should actually have no effect on functionality
```

Comment:
I understand that the order is important as we rely on the `stored name`, right?

Reply:
the structure of the dict we hold the additional names is
```
{
  "stored_name": ["hostname", "shortname", "fqdn"]
}
```
what we end up doing is checking if the name is in the values and if it is, we know the key is the "stored name". So what order in the list of values the hostnames are (e.g. id it was `["fqdn", "hostname", "shortname"]`) we are still just checking against all the values so it wouldn't make a difference. I'll add here that by "stored name" we mean the name used for the dict key in the inventory to get the host's hostspec.

Reply:
I see, thanks for the detailed explanation.